js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerSidebar.tsx

```diff
@@ -37,30 +48,41 @@ export const AssetGraphExplorerSidebar = React.memo(
     explorerPath,
     onChangeExplorerPath,
     allAssetKeys,
+    hideSidebar,
   }: {
     assetGraphData: GraphData;
     lastSelectedNode: GraphNode;
     selectNode: (e: React.MouseEvent<any> | React.KeyboardEvent<any>, nodeId: string) => void;
     explorerPath: ExplorerPath;
     onChangeExplorerPath: (path: ExplorerPath, mode: 'replace' | 'push') => void;
     allAssetKeys: AssetKey[];
+    hideSidebar: () => void;
   }) => {
     const selectNode: typeof _selectNode = (e, id) => {
       _selectNode(e, id);
       if (!assetGraphData.nodes[id]) {
-        const path = JSON.parse(id);
-        const nextOpsQuery = `${explorerPath.opsQuery} \"${path[path.length - 1]}\"`;
-        onChangeExplorerPath(
-          {
-            ...explorerPath,
-            opsQuery: nextOpsQuery,
-          },
-          'push',
-        );
+        try {
+          const path = JSON.parse(id);
+          const nextOpsQuery = `${explorerPath.opsQuery} \"${path[path.length - 1]}\"`;
+          onChangeExplorerPath(
+            {
+              ...explorerPath,
+              opsQuery: nextOpsQuery,
+            },
+            'push',
+          );
+        } catch (e) {
+          // Ignore errors. The selected node might be a group or code location so trying to JSON.parse the id will error.
+          // For asset nodes the id is always a JSON array
```

Comment:
This feels kinda gross - Maybe code locations and groups should not be calling `selectNode`?

Reply:
I still want them to call selectNode, I can use an `if` statement to filter them out but figured try/catch was good enough